torch/csrc/jit/passes/constant_propagation.cpp

```diff
@@ -20,6 +20,11 @@ std::unordered_set<Symbol> skip_list = {
   //FIXME Same problem as in DCE - cpp & python PythonOp and CppOp should be
   //FIXME treated as having side effects but ONNX depends on them being removed
   prim::Print,
+  // A memory fence implies a value is being mutated, so constant propagation
+  // must be disallowed.
```

Comment:
P59934503 is an example that gets tripped up on constant prop.

Basically, if you have an example like this:
```
a = [0, 1]
a.append(2)
a.append(3)
return a == [0, 1, 2, 3]
```
the constant value of `a` will get propagated to the memory fence before both appends, leading to the first append getting "dropped".

Reply:
We could conceivably propagate the constant to the first fence and disallow after, but it's more intrusive to the constant prop pass for little benefit.

Reply:
Actually, I realize now that this is because of the bug you pointed out below, where we re-use the same list as the input to every begin-fence. Will fix!!

Reply:
List allocation is a side effect, it needs to be serialized with world tokens.